### PR TITLE
fix(Table) clean up 'Expand collapse all' demo

### LINKED ISSUE #8976: Table - clean up 'Expand collapse all' demo
Update the demo to use the Table (formerly TableComposable) and not the deprecated Table components.

Also fix TypeError: Cannot read properties of null (reading 'length') when opening the demo, plus the same error for reading 'header' in console

### PR DESCRIPTION
**Pull request by Dominik-Petrik**
**What**: Closes #8976 

I have also revamped the demo to use functional instead of class component.

### CODE REVIEW

**Review by nicolethoen — CHANGES_REQUESTED** [review]
I'm also seeing the codesandbox throw an error about the onPageResize not being defined for the dashboardwrapper. i'm also seeing the same issue in other demos in the codesandbox, though so that can be addressed in the future.

File: packages/react-table/src/docs/demos/table-demos/ExpandCollapseAll.jsx [hunk 1]
```diff
@@ -121,101 +112,89 @@ const serverData = [
...
-    this.setState(() => {
-      const otherExpandedServerNames = this.state.expandedServerNames.filter(r => r !== server.name);
-      return {
-        expandedServerNames: isExpanding ? [...otherExpandedServerNames, server.name] : otherExpandedServerNames
-      };
-    });
+  const setServerExpanded = (server, isExpanding) => {
+    const otherExpandedServerNames = expandedServerNames.filter((r) => r !== server.name);
+    setExpandedServerNames(isExpanding ? [...otherExpandedServerNames, server.name] : otherExpandedServerNames);
+    setAreAllExpanded(expandedServerNames.length == serverData.length);
```
[the remark this answers is not in the document]
Comment (kmcfaul) [reply packages/react-table/src/docs/demos/table-demos/ExpandCollapseAll.jsx:123]: Adding onto this, can we also check whether the `expandedServerNames` length becomes 0? Currently if you expand a row and then collapse it such that there are no expanded rows, the overall "all expanded" carat isn't updating.

**Review by gitdallas — APPROVED** [review]

**Review by kmcfaul — COMMENTED** [review]
The state properly updates when a single folder collapse that results in all folders being collapsed updates the bulk select carat, but seeing a bug where expanding/collapsing a second folder is also toggling the bulk select.

**Comment by gitdallas** [comment]
@kmcfaul - ah that's interesting. i do notice that it is using the state immediately after setting it. it should use the data that is being used to set the state instead of the state itself... or better yet - an effect.

@Dominik-Petrik

**Review by nicolethoen — APPROVED** [review]

**Review by kmcfaul — COMMENTED** [review]
Do we want the bulk select carat to show as "expanded" when some of the folders are expanded? Right now it's only showing "expanded" when all of the folders are open.

If so then it looks good to me

**Comment by nicolethoen** [comment]
I think that top carat mean's expand all - so it should be open only when all are expanded?

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**